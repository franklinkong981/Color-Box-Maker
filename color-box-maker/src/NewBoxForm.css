.NewBoxForm-height-field, .NewBoxForm-width-field {
  width: 50vw;
}