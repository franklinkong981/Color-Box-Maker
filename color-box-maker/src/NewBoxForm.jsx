import React from "react";
import {useFormik} from 'formik';

import "./NewBoxForm.css";

const colors = [
  "aqua", "black", "blue", "brown", "chartreuse", "chocolate", "crimson", "cyan", "fuchsia", "gold", "gray", "green",
  "indigo", "ivory", "lavender", "lime", "magenta", "maroon", "navy", "olive", "orange", "pink", "plum", "purple", "red",
  "salmon", "teal", "turquoise", "violet", "white", "yellow" 
]

const NewBoxForm = ({addBox}) => {
  const formik = useFormik({
    initialValues:{
      color: 'white',
      height: 10,
      width: 10
    },
    onSubmit: values => {
      addBox({color: values.color, height: values.height, width: values.width})
    }
  });

  return (
    <form className="NewBoxForm" onSubmit={formik.handleSubmit}>
      <label htmlFor="color-field">Select Color:</label>
      <select id="color-field" className="NewBoxForm-color-field" name="color" value={formik.values.color} onChange={formik.handleChange}>
        {colors.map(color => (
          <option value={color}>{color.charAt(0).toUpperCase() + color.slice(1)}</option>
        ))}
      </select><br/>

      <label htmlFor="height-field">Height (pixels):</label>
      <input id="height-field" className="NewBoxForm-height-field" type="range" min="10" max="500" name="height"
      value={formik.values.height} onChange={formik.handleChange}/>
      <span className="NewBoxForm-height-display">{formik.values.height}</span><br/>

      <label htmlFor="width-field">Width (pixels): </label>
      <input id="width-field" className="NewBoxForm-width-field" type="range" min="10" max="500" name="width" 
      value={formik.values.width} onChange={formik.handleChange}/>
      <span className="NewBoxForm-width-display">{formik.values.width}</span><br/>

      <button type="submit">Add box</button>
    </form>
  );
};

export default NewBoxForm;