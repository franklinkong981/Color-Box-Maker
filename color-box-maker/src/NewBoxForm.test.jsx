import React from "react";
import {render} from "@testing-library/react";
import NewBoxForm from "./NewBoxForm.jsx";
import { expect, it } from "vitest";

const addBox = () => {
  console.log("Box added");
}

it("renders without crashing", function() {
  render(<NewBoxForm addBox={addBox}/>);
});

it("matches snapshot", function() {
  const {asFragment} = render(<NewBoxForm addBox={addBox}/>);
  expect(asFragment()).toMatchSnapshot();
});