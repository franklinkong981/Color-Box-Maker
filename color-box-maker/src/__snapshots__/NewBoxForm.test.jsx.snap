// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewBoxForm"
  >
    <label
      for="color-field"
    >
      Select Color:
    </label>
    <select
      class="NewBoxForm-color-field"
      id="color-field"
      name="color"
    >
      <option
        value="aqua"
      >
        Aqua
      </option>
      <option
        value="black"
      >
        Black
      </option>
      <option
        value="blue"
      >
        Blue
      </option>
      <option
        value="brown"
      >
        Brown
      </option>
      <option
        value="chartreuse"
      >
        Chartreuse
      </option>
      <option
        value="chocolate"
      >
        Chocolate
      </option>
      <option
        value="crimson"
      >
        Crimson
      </option>
      <option
        value="cyan"
      >
        Cyan
      </option>
      <option
        value="fuchsia"
      >
        Fuchsia
      </option>
      <option
        value="gold"
      >
        Gold
      </option>
      <option
        value="gray"
      >
        Gray
      </option>
      <option
        value="green"
      >
        Green
      </option>
      <option
        value="indigo"
      >
        Indigo
      </option>
      <option
        value="ivory"
      >
        Ivory
      </option>
      <option
        value="lavender"
      >
        Lavender
      </option>
      <option
        value="lime"
      >
        Lime
      </option>
      <option
        value="magenta"
      >
        Magenta
      </option>
      <option
        value="maroon"
      >
        Maroon
      </option>
      <option
        value="navy"
      >
        Navy
      </option>
      <option
        value="olive"
      >
        Olive
      </option>
      <option
        value="orange"
      >
        Orange
      </option>
      <option
        value="pink"
      >
        Pink
      </option>
      <option
        value="plum"
      >
        Plum
      </option>
      <option
        value="purple"
      >
        Purple
      </option>
      <option
        value="red"
      >
        Red
      </option>
      <option
        value="salmon"
      >
        Salmon
      </option>
      <option
        value="teal"
      >
        Teal
      </option>
      <option
        value="turquoise"
      >
        Turquoise
      </option>
      <option
        value="violet"
      >
        Violet
      </option>
      <option
        value="white"
      >
        White
      </option>
      <option
        value="yellow"
      >
        Yellow
      </option>
    </select>
    <br />
    <label
      for="height-field"
    >
      Height (pixels):
    </label>
    <input
      class="NewBoxForm-height-field"
      id="height-field"
      max="500"
      min="10"
      name="height"
      type="range"
      value="10"
    />
    <span
      class="NewBoxForm-height-display"
    >
      10
    </span>
    <br />
    <label
      for="width-field"
    >
      Width (pixels):
    </label>
    <input
      class="NewBoxForm-width-field"
      id="width-field"
      max="500"
      min="10"
      name="width"
      type="range"
      value="10"
    />
    <span
      class="NewBoxForm-width-display"
    >
      10
    </span>
    <br />
    <button
      type="submit"
    >
      Add box
    </button>
  </form>
</DocumentFragment>
`;
